packages/nuxt/src/core/templates.ts

```diff
@@ -157,36 +157,44 @@ const adHocModules = ['router', 'pages', 'imports', 'meta', 'components', 'nuxt-
 export const schemaTemplate: NuxtTemplate = {
   filename: 'types/schema.d.ts',
   getContents: async ({ nuxt }) => {
-    const moduleInfo = nuxt.options._installedModules.map(m => ({
-      ...m.meta,
-      importName: m.entryPath || m.meta?.name,
-    })).filter(m => m.configKey && m.name && !adHocModules.includes(m.name))
-
     const relativeRoot = relative(resolve(nuxt.options.buildDir, 'types'), nuxt.options.rootDir)
     const getImportName = (name: string) => (name[0] === '.' ? './' + join(relativeRoot, name) : name).replace(/\.\w+$/, '')
-    const modules = moduleInfo.map(meta => [genString(meta.configKey), getImportName(meta.importName), meta])
+
+    const modules = nuxt.options._installedModules
+      .filter(m => m.meta && m.meta.configKey && m.meta.name && !adHocModules.includes(m.meta.name))
+      .map(m => [genString(m.meta.configKey), getImportName(m.entryPath || m.meta.name), m] as const)
+
     const privateRuntimeConfig = Object.create(null)
     for (const key in nuxt.options.runtimeConfig) {
       if (key !== 'public') {
         privateRuntimeConfig[key] = nuxt.options.runtimeConfig[key]
       }
     }
     const moduleOptionsInterface = [
-      ...modules.map(([configKey, importName]) =>
+      ...modules.flatMap(([configKey, importName, mod]) => {
+        return [
+        `    /**`,
+        `     * Configuration for \`${importName}\``,
+        mod.meta?.rawPath
+          ? undefined
+          : `     * @see https://www.npmjs.com/package/${importName}`,
```

Comment:
IMO this should be properly inferred from `reponsitory` field or module meta if we are going to generate docs. Many misleading cases can happen by assuming import name is always from npm registry...

Reply:
Yeah. That's why I am bringing up the question above. I see currently those fields are "agnositic" to Nuxt and module authors - not everyone merging their package.json into the meta field, and it could probably be namespace collisions if we always ask module authors to do so. Have a few more explicit field might make this behaviour more clear IMO